src/main.c

```diff
@@ -38,12 +39,38 @@ void SysTick_Handler(void) {
   msTicks++;
 }
 
+void TIMER32_0_IRQHandler(void) {
```

Comment:
only concern is whether these variables need to be marked volatile because they are being updated in an interrupt handler. i think they actually might need to be in which case you should instead have two volatile global variables that contain this informations